### ISSUE TITLE
The remote tree inspector is choking on huge amount of nodes, even when not expanded

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#8165]
Tested on Godot 3.0 af282b56feb8d84049f911dd1d91d82bd2211727

I have a scene containing a procedurally-generated terrain, which generates a lot of nodes under a specific parent.
When the remote tree inspector is enabled, it immediately freezes the editor and the game for a few seconds and repeats regularly, even if the parent holding the huge amount of nodes is not expanded.

It would be nice if the remote inspector didn't try to get all nodes in the scene at once, but only those it needs.

**Comment by akien-mga** [comment]
Can anyone still reproduce this bug in [Godot 3.2.3 rc5](https://godotengine.org/article/release-candidate-godot-3-2-3-rc-5) or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by Zireael07** [comment]
I'm not sure an MRP could be made, since it requires a large amount of nodes.
That said, remote tree inspector performance on a large project is still an issue, even if seems to have improved since @Zylann made the issue.

**Comment by Zylann** [comment]
Tested instancing plenty of nodes. The initial load freezes the game for a second, and then continues normally for a few seconds. Then freezes again, every 10 seconds.

The fact nodes not being expanded still affects that time is hard to test because it keeps expanding the same node on further sessions even if I didn't ask it to do so.

The project I used:
[ManyNodes.zip](https://github.com/godotengine/godot/files/5177313/ManyNodes.zip)
